gtfs/__main__.py

```diff
@@ -1,25 +1,46 @@
 #!/usr/bin/env python
 """Command line interface for fetching GTFS."""
-import argparse
-import getpass
+import typer
+from typing_extensions import Annotated
+from chalky import chain
 import logging
-import sys
-
-from prettytable import PrettyTable
-
-from gtfs.utils.FeedSource import FeedSource
-import feed_sources
-# import all the available feed sources
-# pylint: disable=I0011,wildcard-import
-from feed_sources import *
+from prettytable.colortable import ColorTable, Themes
+from halo import Halo
+from .feed_source import FeedSource
+from .feed_sources import __all__ as feed_sources
 
 logging.basicConfig()
 LOG = logging.getLogger()
 LOG.setLevel(logging.INFO)
+app = typer.Typer()
+            
+@app.command()
+def list_feeds(bbox: Annotated[str, typer.Argument(help="pass value as a string separated by commas like this: xmin,ymin,xmax,ymax ")],
+    predicate: Annotated[str, typer.Option(help="intersects or contains")] = 'intersects'):
+    """Filter feeds spatially based on bounding box."""
+    spinner = Halo(text='Filtering...', text_color="cyan", spinner='dots')
+    spinner.start()
+    xmin, ymin, xmax, ymax = [float(coord) for coord in bbox.split(',')]
```

Comment:
this would need to be protected and further checked. since the input is a string, we don't know yet if it's comma-separated or even if it is comma-separated, if it's really 4 values. in which case it'd fail and we need to make sure that the user gets appropriate feedback (e.g. LOG.critical() & sys.exit(1)).

Reply:
Yeah u are right..I will add a check for that!